lib/datadog/appsec/response.rb

```diff
@@ -67,6 +74,37 @@ def content_type(env)
         rescue Datadog::AppSec::Utils::HTTP::MediaRange::ParseError
           DEFAULT_CONTENT_TYPE
         end
+
+        def status
+          Datadog.configuration.appsec.block.status
+        end
+
+        def redirect?
+          REDIRECT_STATUS.include?(status)
+        end
+
+        def location
+          Datadog.configuration.appsec.block.location
+        end
+
+        def content(content_type)
+          setting = Datadog.configuration.appsec.block.templates[content_type]
+
+          case setting
+          when :html, :json, :text
+            Datadog::AppSec::Assets.blocked(format: setting)
+          when String, Pathname
+            path = setting.to_s
+
+            cache[path] ||= (File.open(path, 'rb', &:read) || '')
+          else
+            raise ArgumentError, "unexpected type: #{content_type.inspect}"
+          end
+        end
+
+        def cache
+          @cache ||= {}
```

Comment:
That cache is in the singleton, but I don't think it's too bad in this case, even when testing.